File: results/dnd-rest-benchmark/lifecycle-runs/20260804T072004Z_claude_claude-sonnet-5_ruby-sinatra/routes/storage.rb
# Storage introspection and a full reset (used by test setup/teardown).

get '/v1/storage/status' do
  { driver: 'sqlite', schema_version: SCHEMA_VERSION, initialized: File.exist?(DB_PATH) }.to_json
end

post '/v1/storage/reset' do
  reset_schema!
  { ok: true, schema_version: SCHEMA_VERSION }.to_json
end

File: results/dnd-rest-benchmark/lifecycle-runs/20260731T062307Z_claude_claude-sonnet-5_typescript-node/src/routes/storage.ts
// Storage introspection and the destructive full-reset used by the test
// suite between scenarios.
import type { ServerResponse } from "node:http";
import { SCHEMA_VERSION, resetDatabase, storageInitialized } from "../db.js";
import { sendJson } from "../http.js";

export function handleStorageStatus(res: ServerResponse): void {
  sendJson(res, 200, {
    driver: "sqlite",
    schema_version: SCHEMA_VERSION,
    initialized: storageInitialized,
  });
}

export function handleStorageReset(res: ServerResponse): void {
  resetDatabase();
  sendJson(res, 200, { ok: true, schema_version: SCHEMA_VERSION });
}
